crates/turbo-tasks-fs/src/invalidation.rs

```diff
@@ -0,0 +1,64 @@
+use std::borrow::Cow;
+
+use indexmap::IndexSet;
+use turbo_tasks::{InvalidationReason, InvalidationReasonType};
+
+pub struct WatchChange {
+    pub path: String,
+}
+
+impl InvalidationReason for WatchChange {
+    fn description(&self) -> Cow<'static, str> {
+        format!("{} changed", self.path).into()
+    }
+    fn merge_info(&self) -> Option<(&'static dyn InvalidationReasonType, Cow<'static, str>)> {
+        Some((&WATCH_CHANGE_TYPE, self.path.clone().into()))
+    }
+}
+
+struct WatchChangeType {
+    _non_zero_sized: u8,
+}
+
+static WATCH_CHANGE_TYPE: WatchChangeType = WatchChangeType { _non_zero_sized: 0 };
+
+impl InvalidationReasonType for WatchChangeType {
+    fn description(&self, merge_data: &IndexSet<Cow<'static, str>>) -> Cow<'static, str> {
+        format!(
+            "{} files changed (e. g. {})",
```

Comment:
There should be two modes:
```
1 file changed ({path})
```
and 
```
N files changed (e.g. {path})
```

Reply:
`InvalidationReasonType::description` will only called for more than 1 merge_data entry.

In the case of a single one, it will use `InvalidationReason::description`.